tcms/testcases/migrations/0010_remove_bug.py

```diff
@@ -1,20 +1,57 @@
+import glob
+import json
+
 from django.db import migrations
 
 
 def forward_copy_data(apps, schema_editor):
     bug_model = apps.get_model('testcases', 'Bug')
     link_reference_model = apps.get_model('linkreference', 'LinkReference')
 
-    for bug in bug_model.objects.all():
-        if not bug.case_run_id:
-            continue
+    link_reference_ids_file_name = '/tmp/kiwitcms-testcases-migrations-0010\
+-new-LinkReference-IDs'  # nosec:B108:hardcoded_tmp_directory
+    with open(link_reference_ids_file_name, 'w') as link_reference_ids_file:
+        for bug in bug_model.objects.all():
+            file_name = '/tmp/kiwitcms-tescases-migrations-\
+0010-Bug-%d' % bug.pk  # nosec:B108:hardcoded_tmp_directory
+            with open(file_name, 'w') as outfile:
+                json.dump(bug.serialize(), outfile)
+
+            if not bug.case_run_id:
+                continue
+
+            link_reference = link_reference_model.objects.create(
+                execution_id=bug.case_run_id,
+                name="%s %s" % (bug.bug_system.name, bug.bug_id),
+                url=bug.bug_system.url_reg_exp % bug.bug_id,
+                is_defect=True,
+            )
+
+            link_reference_ids_file.write('{},'.format(link_reference.id))
+
+
+def backward_restore_data(apps, schema_editor):
+    bug_model = apps.get_model('testcases', 'Bug')
+    link_reference_model = apps.get_model('linkreference', 'LinkReference')
+
+    for file_name in glob.glob(
+            '/tmp/kiwitcms-tescases-migrations-0010-Bug-*'  # nosec:B108:hardcoded_tmp_directory
+    ):
+        with open(file_name, 'r') as infile:
+            data = json.loads(infile)
+            bug = bug_model(**data)
+            bug.save()
 
-        link_reference_model.objects.create(
-            execution_id=bug.case_run_id,
-            name="%s %s" % (bug.bug_system.name, bug.bug_id),
-            url=bug.bug_system.url_reg_exp % bug.bug_id,
-            is_defect=True,
-        )
+    link_reference_ids_file_name = '/tmp/kiwitcms-testcases-migrations-0010-\
+new-LinkReference-IDs'  # nosec:B108:hardcoded_tmp_directory
+    with open(link_reference_ids_file_name, 'r') as link_reference_ids_file:
+        link_reference_ids = link_reference_ids_file.read()
+        for link_reference_id in link_reference_ids[:-1].split(','):
+            try:
+                link_reference = link_reference_model.objects.get(pk=link_reference_id)
+                link_reference.delete()
+            except link_reference_model.DoesNotExist:
+                print('LinkReference with id {} not found'.format(link_reference_id))
```

Comment:
Once all of the ids are loaded you can use `.filter(pk__in=list_of_ids).delete()` to remove everything in one operation.